### PR TITLE
feat: add selector card component

### PR DESCRIPTION
**Pull request by BartDeCaluwe**
This PR will:

- add SelectorCard component which is a radio button with supports rich content

## Checklist

- [x] I used the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) specification.
- [ ] Unit tests have been created if a new feature was added.
- [ ] Translations have been added and extracted if a new feature was added.

### CODE REVIEW

**Review by BouweCeunen — DISMISSED** [review]

File: src/components/BulletLoanRepaymentSchedule.tsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+		type: RepaymentScheduleAmountType.PRINCIPAL,
+		amount: 90,
+	});
+	data.push({
+		month: "12",
+		type: RepaymentScheduleAmountType.INTEREST,
+		amount: 10,
+	});
+
+	const dataSource = [
```
Comment (RuvendeGroote) [inline src/components/BulletLoanRepaymentSchedule.tsx]: remove this testing data

File: src/components/RepaymentScheduleTable.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+export interface RepaymentScheduleDataPoint {
+	month: string;
+	type: RepaymentScheduleAmountType;
+	amount: number;
+}
+
+interface RepaymentScheduleTableProps {
+	dataSource: AntdTableProps<any>["dataSource"];
```
Comment (RuvendeGroote) [inline src/components/RepaymentScheduleTable.tsx]: Use the props of our own table

Comment (RuvendeGroote) [reply src/components/RepaymentScheduleTable.tsx]: We can also avoid the any here as we know what kind of data we will be representing here. We can also avoid it in our Table by making the component generic

File: src/components/BulletLoanRepaymentSchedule.tsx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+interface BulletLoanRepaymentScheduleProps {
+	children?: ReactNode;
+}
+export const BulletLoanRepaymentSchedule: FunctionComponent<
+	BulletLoanRepaymentScheduleProps
+> = () => {
+	const [showTable, setShowTable] = useState(false);
+
+	const data: RepaymentScheduleDataPoint[] = [];
```
Comment (RuvendeGroote) [inline src/components/BulletLoanRepaymentSchedule.tsx]: this also looks like test data. remove it.

File: src/components/RepaymentScheduleGraph.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import { Column } from "@ant-design/plots";
+import { FunctionComponent } from "react";
+
+export enum RepaymentScheduleAmountType {
+	INTEREST = "interest",
+	PRINCIPAL = "principal",
+}
+
+export interface RepaymentScheduleDataPoint {
```
Comment (RuvendeGroote) [inline src/components/RepaymentScheduleGraph.tsx]: this interface type is duplicate in `RepaymentScheduleTable`

File: src/components/RepaymentScheduleGraph.tsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+		data,
+		height: 200,
+		width: 320,
+		isStack: true,
+		xField: "month",
+		yField: "amount",
+		seriesField: "type",
+		legend: false as unknown as object,
+		colorField: "type",
+		color: ["#D8DEDA", "#718879"],
```
Comment (RuvendeGroote) [inline src/components/RepaymentScheduleGraph.tsx]: use colors from theme.js

File: src/components/RepaymentScheduleGraph.tsx [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+export const RepaymentScheduleGraph: FunctionComponent<RepaymentScheduleGraphProps> = ({
+	data,
+}) => {
+	const config = {
+		data,
+		height: 200,
+		width: 320,
+		isStack: true,
+		xField: "month",
```
Comment (RuvendeGroote) [inline src/components/RepaymentScheduleGraph.tsx:17]: use intl

File: src/components/AmortizationRepaymentSchedule.tsx [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+} from "./RepaymentScheduleGraph";
+import { RepaymentScheduleTable } from "./RepaymentScheduleTable";
+import { Button } from "./Button";
+import { ColumnsProps } from "./Table";
+
+interface AmortizationRepaymentScheduleProps {
+	children?: ReactNode;
+}
+
+export const AmortizationRepaymentSchedule: FunctionComponent<
```
Comment (RuvendeGroote) [inline src/components/AmortizationRepaymentSchedule.tsx:16]: we should receive information like principal, interest and duration (and/or others) and generate the schedule dynamically based on that information. this is just displaying static test data now. same with bulletloan

**Review by RuvendeGroote — CHANGES_REQUESTED** [review]

File: src/components/AmortizationRepaymentSchedule.tsx (hunk 7 above)
Comment (RuvendeGroote) [reply src/components/AmortizationRepaymentSchedule.tsx:16]: I noticed we have separate tickets for that stuff so you can disregard this comment

File: src/components/RepaymentScheduleGraph.tsx (hunk 6 above)
Comment (BartDeCaluwe) [reply src/components/RepaymentScheduleGraph.tsx:17]: The "month" refers to an object property, `intl` can't be used in this context.

File: src/components/RepaymentScheduleTable.tsx [hunk 8]
```diff
@@ -0,0 +1,22 @@
+import React, { FunctionComponent } from "react";
+import { ColumnsProps, Table, TableProps } from "@components/Table";
+import { Button } from "@components/Button";
+
+interface RepaymentScheduleTableProps {
+	dataSource: TableProps["dataSource"];
```
Comment (RuvendeGroote) [inline src/components/RepaymentScheduleTable.tsx]: this is basically any but we know what data we are expecting here

**Review by RuvendeGroote — CHANGES_REQUESTED** [review]

**Review by RuvendeGroote — APPROVED** [review]

**Pull request merged**